### ISSUE TITLE
catch TypeError in sync call

### ISSUE THREAD

**Issue by ffoysal** [qunitjs/qunit#1062]
I have tests in two files, one file

``` js
QUnit.test( "Test 1", function( assert ) {

    var obj = undefined;
    var doneFun = assert.async();
    setTimeout(function(){
        assert.equal('hell', obj.hell, 'Hello');
        doneFun();
    },100) });
```

and the other is 

``` js
QUnit.test( "Test 2", function( assert ) {
    var doneFun = assert.async();
    setTimeout(function(){
        assert.equal('hell', 'hell','hello');
        doneFun();
    },500); });
```

when I run both together I got

```
        assert.equal('hell', obj.hell, 'Hello');
                                ^

TypeError: Cannot read property 'hell' of undefined
    at null._onTimeout (/home/foysal/workspace/qa/qa-test-framework/source/tests/mytest.js:12:30)
    at Timer.listOnTimeout (timers.js:92:15)
```

want to catch any TypeError in async test.
I am running in nodejs.

Help is very much appreciated

**Comment by leobalter** [comment]
> Help is very much appreciated

Likewise, always.

It's interesting this bug happens on the git version, but the current released version still catches the error: http://jsbin.com/jevutajuni/edit?html,js,console,output (using QUnit 2.0.1)

<img width="472" alt="screenshot 2016-10-17 18 46 48" src="https://cloud.githubusercontent.com/assets/301201/19458440/1774a4c2-949a-11e6-9ad9-734667e4477a.png">

The current git version from the master branch is not catching the exception:

<img width="470" alt="screenshot 2016-10-17 18 47 44" src="https://cloud.githubusercontent.com/assets/301201/19458466/389b4eda-949a-11e6-8871-dd08d5d37d43.png">

---

I'll run a git bisect to find this bug and fix it. Thanks for the report.

**Comment by leobalter** [comment]
oh, this is an uncaught bug from 80a14ee5ac49c7f0981884a6cbb5786dea3a9c8b (PR #1025).

Now it's time to figure out how to fix it.

**Comment by leobalter** [comment]
> I am running in nodejs.

Ok, that's another issue we need to address. 2 bugs in 1 issue. 👍

**Comment by trentmwillis** [comment]
Closing this as the behavior shown by @leobalter is fixed in `2.1.0` and on the current `master` branch (either due to #1063 or as a side-effect of something else).

As for catching uncaught exceptions, as I believe is the original request, you can define your own `window.onerror` handler in the Browser or your own `process.on('uncaughtException')` handler in Node. While we need to add a solution for uncaught errors in Node by default, that should be tracked in https://github.com/qunitjs/qunit/issues/1061.
